File: binarySearch.js
function binarySearch(array, target) {	
	return binarySearchHelper(array, target, 0, array.length - 1);	
};

function binarySearchHelper(array, target, left, right) {
	if (left > right) return -1;
	const midIndex = Math.floor((left + right) / 2)
	const midpoint = array[midIndex]	
	if (midpoint === target) return midIndex;
	if (target < midpoint) {
		return binarySearchHelper(array, target, left, midIndex - 1);
	} 
	if (target > midpoint) {
		return binarySearchHelper(array, target, midIndex + 1, right);
	}	
}

File: binary_search.py
import math


def binarySearch(array, target):
    left = 0
    right = len(array) - 1

    while left <= right:
        middle = (left + right) // 2
        match = array[middle]
        if match == target:
            return middle
        elif match > target:
            right = middle - 1
        else:
            left = middle + 1
    return -1


# Implement Binary Search
# Initialize left pointer at idx zero
# Initialize right pointer at idx len(array) - 1
# Initialize middle (left, right)/2 rounded down
# while left < right
# If middle == target, return middle
# IF middle > target, right = middle - 1
# If middle < target left = middle + 1
# return - 1
